unit_tests/fixtures/raster_data.asc

```diff
@@ -0,0 +1,10 @@
+0  0  0  0  0  2  4  6  8  10
+0  0  0  0  2  4  6  8  10 12
+0  0  0  2  4  6  8  10 12 14
+0  0  2  4  6  8  10 12 14 16
+0  2  4  6  8  10 12 14 16 18
+0  0  2  4  6  8  10 12 14 16
+0  0  0  2  4  6  8  10 12 14
+0  0  0  0  2  4  6  8  10 8
+0  0  0  0  0  2  4  6  8  6
+0  0  0  0  0  0  2  4  6  4
```

Comment:
I would change the edge values to something different than `0`. This way when your grid subscript `operator()()` does an out of bounds access and does not get killed but accesses memory to the left/right instead which may be zeroed out, you can detect this more easily.